domain/consensus/processes/dagtraversalmanager/block_heap.go

```diff
@@ -33,34 +37,39 @@ func (h *baseHeap) Pop() interface{} {
 	return popped
 }
 
+// peek returns the block with lowest height from this heap without removing it
+func (h *baseHeap) peek() *blockHeapNode {
+	return h.slice[0]
```

Comment:
Is it guaranteed to be the first one?

Reply:
Yes. https://golang.org/pkg/container/heap/
> The minimum element in the tree is the root, at index 0.